did/validator.go

```diff
@@ -182,6 +185,12 @@ func (s serviceValidator) Validate(document Document) error {
 		default:
 			return makeValidationError(ErrInvalidService)
 		}
+		// service ID must be unique
+		for x, other := range document.Service {
+			if i != x && service.ID == other.ID {
```

Comment:
small optimization? (if it doesn't copy the sub slice)
```suggestion
		for _, other := range document.Service[i+1:] {
			if service.ID == other.ID {
```

Reply:
Neat idea. I think slices of slices in Go are really cheap, since they are basically pointers to the other slice.